main.js

```diff
@@ -4666,8 +4682,31 @@ function init(compactGroupId) {
             states.subscribe('system.adapter.*.alive');
 
             // set current Loglevel and subscribe for changes
-            states.setState(hostObjectPrefix + '.logLevel', {val: config.log.level, ack: true, from: hostObjectPrefix});
-            states.subscribe(hostObjectPrefix + '.logLevel');
+            states.setState(`${hostObjectPrefix}.logLevel`, {val: config.log.level, ack: true, from: hostObjectPrefix});
+            states.subscribe(`${hostObjectPrefix}.logLevel`);
+
+            try {
+                const nodeVersion = process.version.replace(/^v/, '');
+                const prevNodeVersionState = await states.getStateAsync(`${hostObjectPrefix}.nodeVersion`);
+
+                if (prevNodeVersionState && prevNodeVersionState.val !== nodeVersion) {
+                    // detected a change in the nodejs version
+                    logger.info(`${hostLogPrefix} Node.js version has changed from ${prevNodeVersionState.val} to ${nodeVersion}`);
+                    if (os.platform() === 'linux') {
+                        // ensure capabilities are set
+                        await tools.setExecutableCapabilities(process.execPath, ['cap_net_admin', 'cap_net_bind_service', 'cap_net_raw'], true, true, true);
+                    }
+                }
+
+                // set current node version
+                await states.setStateAsync(`${hostObjectPrefix}.nodeVersion`, {
+                    val: nodeVersion,
+                    ack: true,
+                    from: hostObjectPrefix
+                });
+            } catch (e) {
+                logger.warn(`${hostLogPrefix} Error on node version check routine: ${e.message}`);
```

Comment:
I would adjust error rmesage like "Error while trying to update  capabilities after updating nodejs: ..." ?!

Reply:
Hm it could also be that the setState for the version throws, but ok for me